include/proxy-wasm/context_interface.h

```diff
@@ -0,0 +1,628 @@
+// Copyright 2016-2019 Envoy Project Authors
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "include/proxy-wasm/compat.h"
+
+#include <time.h>
+#include <atomic>
+#include <chrono>
+#include <functional>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <vector>
+
+namespace proxy_wasm {
+
+#include "proxy_wasm_common.h"
+#include "proxy_wasm_enums.h"
+
+using Pairs = std::vector<std::pair<string_view, string_view>>;
+using PairsWithStringValues = std::vector<std::pair<string_view, std::string>>;
+using HttpCallToken = uint32_t;
+using GrpcToken = uint32_t;
+using SharedQueueDequeueToken = uint32_t;
+using SharedQueueEnqueueToken = uint32_t;
+
+// TODO: move to SDK.
+enum WasmStreamType {
+  Request,
+  Response,
+  Downstream,
+  Upstream,
```

Comment:
This seems like apples/oranges without further comment, i.e. how is request an alternative to downstream?